docs/understanding-docker.md

```diff
@@ -17,273 +17,282 @@ weight = -90
+the foreground in interactive mode (`-i`), and to run the `/bin/bash` command.
 
     $ docker run -i -t ubuntu /bin/bash
 
-The Docker Engine client is launched using the `docker` binary with the `run` option
-running a new container. The bare minimum the Docker client needs to tell the
-Docker daemon to run the container is:
 
-* What Docker image to build the container from, for example, `ubuntu`
-* The command you want to run inside the container when it is launched,
-for example,`/bin/bash`
+When you run this command, Docker Engine does the following:
+
+1. **Pulls the `ubuntu` image:** Docker Engine checks for the presence of the
+    `ubuntu` image. If the image already exists locally, Docker Engine uses it for
+    the new container. Otherwise, then Docker Engine pulls it from
+    [Docker Hub](https://hub.docker.com).
+
+1. **Creates a new container:** Docker uses the image to create a container.
+
+1. **Allocates a filesystem and mounts a read-write _layer_:** The container is
+    created in the file system and a read-write layer is added to the image.
+
+1. **Allocates a network / bridge interface:** Creates a network interface that
+    allows the Docker container to talk to the local host.
 
-So what happens under the hood when we run this command?
+1. **Sets up an IP address:** Finds and attaches an available IP address from a
+    pool.
 
-In order, Docker Engine does the following:
+1. **Executes a process that you specify:** Executes the `/bin/bash` executable.
 
-- **Pulls the `ubuntu` image:** Docker Engine checks for the presence of the `ubuntu`
-image. If the image already exists, then Docker Engine uses it for the new container.
-If it doesn't exist locally on the host, then Docker Engine pulls it from
-[Docker Hub](https://hub.docker.com).
-- **Creates a new container:** Once Docker Engine has the image, it uses it to create a
-container.
-- **Allocates a filesystem and mounts a read-write _layer_:** The container is created in
-the file system and a read-write layer is added to the image.
-- **Allocates a network / bridge interface:** Creates a network interface that allows the
-Docker container to talk to the local host.
-- **Sets up an IP address:** Finds and attaches an available IP address from a pool.
-- **Executes a process that you specify:** Runs your application, and;
-- **Captures and provides application output:** Connects and logs standard input, outputs
-and errors for you to see how your application is running.
+1. **Captures and provides application output:** Connects and logs standard input,
+    outputs and errors for you to see how your application is running, because you
+    requested interactive mode.
 
-You now have a running container! Now you can manage your container, interact with
-your application and then, when finished, stop and remove your container.
+After the container is running, you can manage and interact with it, and eventually
+stop and remove it.
```

Comment:
Grammar nitpick, if you switch to "After the container is running, ..." the subject should agree (e.g., after the container is running, it will ...). But more so, (a) it's nice to get a little success message, and (b) although it's wordy, for this intro we might need to keep the clear distinction between the container, the "application" it's hosting, and your interactions it both. 

How about: 

You now have a running container! You can manage the container externally, interact directly with the application it is serving up  (in this case, `ubuntu`), and eventually stop and remove the container.